src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs

```diff
@@ -92,6 +92,31 @@ CandidateReason.Ambiguous or
                 return true;
             }
 
+            if (name is IdentifierNameSyntax { Identifier.Text: "async" } && symbol is null)
```

Comment:
❓ This currently appears to be on the semantic classifier path (where `symbol` exists). Can we move it to the syntax classifier path?

Reply:
I am a bit confused now. I need to know, if `async` was binded to any existing symbol and in general the initial goal of the PR was to colorize `async` in cases, where parser recognize it as an identifier, so It should be in `NameClassifier`, right?